### PR TITLE
stdatomic.h / atomic compatibility for C/C++ 11 standard

### PR DESCRIPTION
**Pull request by breznak**
required for g++4.9 (maybe others), as explained in https://gcc.gnu.org/bugzilla/show_bug.cgi?id=60932

### CODE REVIEW

File: utils/tcpststat/tcpststat.c [hunk 1]
```diff
@@ -13,7 +13,6 @@
 #include <libcaer/events/frame.h>
 
 #include <signal.h>
-#include <stdatomic.h>
```
Comment (breznak) [inline utils/tcpststat/tcpststat.c]: already covered in `inout_common.h` , in other utils too..

**Comment by llongi** [comment]
Okay, let's start from the beginning: what is the problem?
Almost all of these files are not intended to be compiled or included by a C++ compiler, ever, so I'm puzzled about the reason for this proposed change.

**Comment by breznak** [comment]
> Almost all of these files are not intended to be compiled or included by a C++ compiler, ever 

Uh, ok.. 

What am I doing? Writing a C++ client/wrapper for networking (basically I include `tcpststats` into C++ code). 

For that I need to include `ext/libuv.h`, which (through some deps) needs `modules/misc/inout_common.h`. 

Ok, in the PR I maybe went too broad, I did a `grep -R stdatomic` and replaced all with the `#ifdef`. Only for inout_common is the minimal change needed. 

Alternative, I could (should) have implemented the client in C++ right off and only call `libcaer`. But I already had it working with the wrapped C-code, and the recent big PR with `libuv` broke it, so I'm fixing that.

**Comment by llongi** [comment]
This has been fixed differently in commit ee362f658affbb24857d79e17ba5511e927d3f0c.
inout_common.h may be useful to include in other project, as it contains base definitions, so I've cleaned up its includes. It doesn't need anything special now. ext/libuv.h on the other hand is not intended to be used in C++ at all, as are any of the other .c files where you proposed a change. libuv is a C only library. If you want to reimplement tcpststat in C++, I highly recommend using actual C++ libraries to do the network communication, like Boost. Then only include libcaer headers, or at most inout_common.h if you need its definitions.

**Comment by breznak** [comment]
Thanks for the change. And yes, that's what I've eventually ended up doing..a rewrite to C++ with calling only `libcaer` is a cleaner solution. Thanks for bearing with me ;)

**Pull request closed without merging**